lib/improver/nowcasting/forecasting.py

```diff
@@ -324,3 +327,101 @@ def grid_spacing(coord):
 
         advected_cube = amend_metadata(advected_cube, **self.metadata_dict)
         return advected_cube
+
+
+class CreateExtrapolationForecast(object):
+    """
+    Class to create a nowcast extrapolation forecast using advection.
+    For precipitation rate forecasts orographic enhancement must be used.
+    """
+
+    def __init__(self, input_cube, vel_x, vel_y,
+                 orographic_enhancement_cube=None, metadata_dict=None):
+        """
+        Initialises the plugin.
+        This includses checking if orographic enhancement is provided and
+        removing the orographic enhancement from the input file ready for
+        extrapolation.
+        An error is raised if the input cube is precipitation rate but no
+        orographic enhancement cube is provided.
+
+        Args:
+            input_cube (iris.cube.Cube):
+                A 2D cube containing data to be advected.
+            vel_x (iris.cube.Cube):
+                Cube containing a 2D array of velocities along the x
+                coordinate axis
+            vel_y (iris.cube.Cube):
+                Cube containing a 2D array of velocities along the y
+                coordinate axis
+
+
+        Keyword Args:
+            orographic_enhancement_cube (iris.cube.Cube):
+                Cube containing the orographic enhancement fields. May have
+                data for multiple times in the cube. This is removed from the
+                input_cube before advecting, and added back on after advection.
+            metadata_dict (Dictionary):
+                Dictionary containing information for amending the metadata
+                of the output cube. Please see the
+                :func:`improver.utilities.cube_metadata.amend_metadata`
+                for information regarding the allowed contents of the metadata
+                dictionary.
+        """
+        self.orographic_enhancement_cube = orographic_enhancement_cube
+        if self.orographic_enhancement_cube:
+            input_cube, = ApplyOrographicEnhancement("subtract").process(
+                input_cube, self.orographic_enhancement_cube)
+        elif "precipitation_rate" in input_cube.name():
+            msg = ("For precipitation fields, orographic enhancement "
+                   "cube must be supplied.")
+            raise ValueError(msg)
+        self.input_cube = input_cube
+        self.advection_plugin = AdvectField(
+            vel_x, vel_y, metadata_dict=metadata_dict)
+
+    def __repr__(self):
+        """Represent the plugin instance as a string."""
+        result = ('<CreateExtrapolationForecast: input_cube = {}, '
+                  'orographic_enhancement_cube = {}, '
+                  'advection_plugin = {}>'.format(
+                      self.input_cube.name(),
```

Comment:
Optional: Could use the cube summary invoked by print(CubeList).

Reply:
Not sure what you mean by this